lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1282,26 +1283,38 @@ WasmBytecodeGenerator::GetRegisterSpace(WasmTypes::WasmType type) const
 EmitInfo
 WasmBytecodeGenerator::PopEvalStack()
 {
-    return m_evalStack->Top()->Pop();
+    // The scope marker should at least be there
+    Assert(!m_evalStack.Empty());
+    EmitInfo info = m_evalStack.Pop();
+    if (info.type == WasmTypes::Limit)
+    {
+        throw WasmCompilationException(_u("Missing operand"));
+    }
+    return info;
 }
 
 void
 WasmBytecodeGenerator::PushEvalStack(EmitInfo info)
 {
-    m_evalStack->Top()->Push(info);
+    Assert(!m_evalStack.Empty());
+    m_evalStack.Push(info);
 }
 
 void
 WasmBytecodeGenerator::EnterEvalStackScope()
 {
-    m_evalStack->Push(Anew(&m_alloc, evalStackScope, &m_alloc));
+    m_evalStack.Push(EmitInfo(WasmTypes::Limit));
```

Comment:
Ah, nice. I was considering introducing a separate scope marker in my first pass instead of a stack of stacks but ended up not wanting to introducing some variation of EmitInfo. EmitInfo(WasmTypes::Limit) is a nice choice.